presto-main/src/main/java/com/facebook/presto/server/WorkerResource.java

```diff
@@ -66,6 +69,44 @@ public Response getThreads(@PathParam("nodeId") String nodeId)
         return proxyJsonResponse(nodeId, "v1/thread");
     }
 
+    @GET
+    public Response getWorkerList()
+    {
+        Set<Node> nodes = nodeManager.getAllNodes().getActiveNodes();
+        Set<JsonNodeInfo> jsonNodes = new HashSet<>();
+        for (Node node : nodes) {
+            JsonNodeInfo jsonNode = new JsonNodeInfo(node.getNodeIdentifier(), node.getHostAndPort().getHostText());
+            jsonNodes.add(jsonNode);
+        }
+        return Response.ok().entity(jsonNodes).build();
+    }
+
+    public static class JsonNodeInfo
```

Comment:
1. See if you can use a private static class.
2. Change the name to NodeInfo. The fact that it is serialized via JSON is an implementation detail.

Reply:
I have changed the name to NodeInfo.
But it looks like can't use  a private static class.